Keep a fallback for the is_reference builtins missing in older g++.

--- a/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/__type_traits/is_reference.h
+++ b/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/__type_traits/is_reference.h
@@ -18,6 +18,8 @@
 
 _LIBCPP_BEGIN_NAMESPACE_STD
 
+#if __has_builtin(__is_reference)
+
 template <class _Tp>
 struct _LIBCPP_NO_SPECIALIZATIONS is_reference : _BoolConstant<__is_reference(_Tp)> {};
 
@@ -26,6 +28,22 @@
 _LIBCPP_NO_SPECIALIZATIONS inline constexpr bool is_reference_v = __is_reference(_Tp);
 #endif
 
+#else // __has_builtin(__is_reference)
+
+template <class _Tp>
+struct _LIBCPP_NO_SPECIALIZATIONS is_reference : public false_type {};
+template <class _Tp>
+struct _LIBCPP_NO_SPECIALIZATIONS is_reference<_Tp&> : public true_type {};
+template <class _Tp>
+struct _LIBCPP_NO_SPECIALIZATIONS is_reference<_Tp&&> : public true_type {};
+
+#if _LIBCPP_STD_VER >= 17
+template <class _Tp>
+_LIBCPP_NO_SPECIALIZATIONS inline constexpr bool is_reference_v = is_reference<_Tp>::value;
+#endif
+
+#endif // __has_builtin(__is_reference)
+
 #if __has_builtin(__is_lvalue_reference) && __has_builtin(__is_rvalue_reference)
 
 template <class _Tp>
